core/modules/redirect/redirect.module

```diff
@@ -1322,9 +1322,9 @@ function redirect_field_attach_form($entity_type, $entity, &$form, &$form_state,
   }
 
   $uri = $entity->uri();
-  if (empty($uri['path'])) {
-    // If the entity has no source path, then we cannot lookup the existing
-    // redirects.
+  if (empty($uri['path']) || substr($uri['path'], -1) == '/') {
```

Comment:
I don't understand why you test for a trailing slash ?

Reply:
@akalata thanks for the PR!  
Code works as advertised.  I see that the `$uri['path']` for a new node is `node/` which is why your code works, but I think it would be better to check if the entity has an `nid` yet.

```php
if (empty($uri['path']) || $entity->nid == NULL) {
```

@akalata you up for updating this?

Reply:
nice catch @serundeputy ; What about using `$entity->isNew()` ? It will work new taxonomy term either.

Reply:
Good call @opi I like using $entity->isNew() better

Reply:
addressed in https://github.com/backdrop/backdrop/pull/2046